src/data/fintype/fin.lean

```diff
@@ -45,4 +47,20 @@ begin
   { rintro ⟨i, hi, rfl⟩, simpa },
 end
 
+lemma card_filter_univ_fin (p : (fin (n + 1)) → Prop) [decidable_pred p] :
+  (univ.filter p).card = (ite (p 0) 1 0) + (univ.filter (p ∘ fin.succ)).card :=
```

Comment:
Do you think this would be better or worse as
```suggestion
  (univ.filter p).card = if p 0 then (univ.filter (p ∘ fin.succ)).card + 1 else (univ.filter (p ∘ fin.succ)).card :=
```
I guess in principle we could have both to match `list.count_cons` vs `list.count_cons'`

Reply:
I've added both with the same `count_cons` naming convention. This seems more generally useful but the other is much cleaner for the other proofs here.